### DISCUSSION TITLE
How to identify a string as a body of a JSON Object?

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#1965] [Q&A]
Here is an open question:
- How to identify a string _(by REGEX, for example)_ as a body of a JSON Object?

Hints:
- _Perhaps with the presence of a key as: `^"string":.*`_ ?

**Comment by arnaudroques** [comment]
Unfortunately, it is not possible using a regular expression.

Identifying a string as a JSON object using regular expressions is inherently problematic due to **the recursive nature of JSON structures**. JSON objects can contain nested objects, arrays, and other complex elements, making it challenging for regular expressions to accurately parse and validate them.

Regular expressions are designed to handle patterns in regular languages, **which do not encompass recursive or nested structures**. As a result, they are not well-suited for parsing languages like JSON that allow for arbitrary levels of nesting. Attempting to use regular expressions for this purpose can lead to incomplete or incorrect matches, especially when dealing with deeply nested or complex JSON data.

For a more in-depth discussion on the limitations of regular expressions in handling nested constructs, you can refer to the article https://www.codeproject.com/Articles/21080/In-Depth-with-RegEx-Matching-Nested-Constructions

**Reply by The-Lum** [reply]
Hi @arnaudroques,

As mentioned on https://github.com/plantuml/plantuml/pull/1968#issuecomment-2456023008
>Thank for the answer but this doesn't help too much.
>I just want to identify that the beginning of the line contains a key of a JSON Object.

Regards,
Th.

**Comment by The-Lum** [comment]
Hi all,

Here is a proposal...:
```java
StringToMatch.matches("^\\s*\"(\\\"|[^\"])*\"\\s*:.*")
```
Open questions:
- Perhaps it will be necessary to replace `\\s` by `\\s\u00A0`?
- Is it also necessary to compile the REGEX or not?
 _(to perf. point of view...)_

Regards,
Th.
